src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -135,22 +135,19 @@ CHIP_ERROR AddCommonTxtElements(const BaseAdvertisingParams<Derived> & params, c
     // change or device type change.
     // TODO: Is this really the best place to set these? Seems like it should be passed
     // in with the correct values and set one level up from here.
-#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    if (chip::DeviceLayer::ConnectivityMgr().GetThreadDeviceType() ==
-        chip::DeviceLayer::ConnectivityManager::kThreadDeviceType_SleepyEndDevice)
-    {
-        uint32_t sedPollPeriod;
-        ReturnErrorOnFailure(chip::DeviceLayer::ThreadStackMgr().GetPollPeriod(sedPollPeriod));
-        // Increment default MRP retry intervals by SED poll period to be on the safe side
-        // and avoid unnecessary retransmissions.
-        if (mrpRetryIntervalIdle.HasValue())
-        {
-            mrpRetryIntervalIdle.SetValue(mrpRetryIntervalIdle.Value() + sedPollPeriod);
-        }
-        if (mrpRetryIntervalActive.HasValue())
-        {
-            mrpRetryIntervalActive.SetValue(mrpRetryIntervalActive.Value() + sedPollPeriod);
-        }
+#if CHIP_DEVICE_CONFIG_ENABLE_SED
+    chip::DeviceLayer::ConnectivityManager::SEDPollingConfig sedPollingConfig;
+    sedPollingConfig.Clear();
+    ReturnErrorOnFailure(chip::DeviceLayer::ConnectivityMgr().GetSEDPollingConfig(sedPollingConfig));
+    // Increment default MRP retry intervals by SED poll period to be on the safe side
+    // and avoid unnecessary retransmissions.
+    if (mrpRetryIntervalIdle.HasValue())
+    {
+        mrpRetryIntervalIdle.SetValue(mrpRetryIntervalIdle.Value() + sedPollingConfig.SlowPollingIntervalMS);
```

Comment:
Should this be an increment, or a max()?

Reply:
That is a good question and it refers also to the MRP interval value itself. In the spec it is said that CRA/CRI value should be the function of SED polling interval, but for me it's not clear enough. What kind of function and who should define it? Constant value is also a mathematical function of SED polling. Also the MRP intervals values are not clear for me, because it may depend for example on how many hops there are in the network between device A and device B. Should device calculate it somehow on its own, or we should assume as big interval to fit most of the cases? Basically I selected here an increment not max value, as poll interval is a time between following device wake-ups and data polls, so in the worst case the time it needs to answer might be bigger than poll interval depending on the network topology. As we discussed with @pan-apple and guys from Infineon there are already few things to fix in the spec regarding SEDs, so we can also consider describing that one better.

Reply:
OK, makes sense.